schemas/data/metrics.schema.json

```diff
@@ -16,65 +16,29 @@
     }
   },
   "properties": {
-    "https://ns.adobe.com/xdm/data/metrics/commerce/abandons": {
```

Comment:
I discussed with (and agree with) @kstreeter that the full URIs here are inconvenient and that we would rather prefer short names. @cmathis: the "storage" argument is not so powerful as the "query language/consumability" argument, btw.

But: I think there are easier ways of doing this than the proposed approach. Let's consider following points:

1. yes, metrics can be grouped, but naming patterns like URIs or prefixes are better than nested objects, because ultimately, the metrics object is just a list of measures.
2. We could just use namespaces instead of full URIs for the known properties: `https://ns.adobe.com/xdm/data/metrics/commerce/abandons` becomes `commerce:abandons` (disadvantage: we have to define more namespaces)
3. We could just use one namespace, and patterns there, such as: `metrics:commerce-abandons` (disadvantage: nesting is less clear)
4. In any case we have to decide if the short name is a replacement or a synonym for the full URI, e.g. if both would be valid. Both approaches come with pros and cons.
5. In any case, `metrics` will be an area where extensibility of the platform will be tested. People will invent new metrics that are not part of XDM Core quickly.

Reply:
We can likely make a smaller namespace such as "metrics" work, but it will change the structure of a schema that has been around for six months and has solutions using it today.  Not sure you are aware of the issue with URI's, but the use of colons and slashes within a field name are not possible within the data storage tech. As a result these longer URI's have to be converted into a nested structure which REALLY changes what we had before. The closer we are to the original schema we are using today the better. The data platform has never had to deal with these long json-ld style schemas which need special translations in and out of the underlying storage tech. As of today, these schemas need to support both the ingestion, storage, and consumption format of the data.   

The "uniqueID" associated with each metric is to support a use-case where the user wants to  associate a value with a given measurement that can be used to identify duplicates later. That value could be a transaction ID, or any value that can be used during query time to remove duplicate records which is a common issue. Without this uniqueID, the query engines have no easy way to filter these duplicates out. We no doubt need a better description on this field.